x-pack/plugins/rule_registry/server/rule_data_client/rule_data_client.ts

```diff
@@ -22,12 +25,21 @@ interface ConstructorOptions {
   isWriteEnabled: boolean;
   waitUntilReadyForReading: Promise<WaitResult>;
   waitUntilReadyForWriting: Promise<WaitResult>;
+  logger: Logger;
 }
 
 export type WaitResult = Either<Error, ElasticsearchClient>;
 
 export class RuleDataClient implements IRuleDataClient {
-  constructor(private readonly options: ConstructorOptions) {}
+  private _isWriteEnabled: boolean = false;
+
+  // Writers cached by namespace
+  private writerCache: Map<string, IRuleDataWriter>;
+
+  constructor(private readonly options: ConstructorOptions) {
+    this.writeEnabled = this.options.isWriteEnabled;
```

Comment:
@Kerry350 Shouldn't it be `options.isWriteEnabled`?

Reply:
As a constructor argument this will be assigned to `this`, so `this.options` is okay.

Reply:
Thanks for clarifying! I knew I was missing something here.